### PR TITLE
Set up skeleton API package

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR sets up {daedalus.api} with some skeleton API code taken from {odin.api}:

1. Updates the `.lintr` file to conform to modern {lintr} and tweak exclusions for;
2. Updates the GHA workflows;
3. Updates tests - removed {odin}-specific tests leaving only basic API tests;
4. Updates the Dockerfile and `Rprofile.site` to be use P3M.

Some outstanding points/issues:

1. I'm not having much luck updating to `rocker/r-ver:4.4.1` as {R6} and {jsonlite} aren't installing which blocks {porcelain};
2. There's little to no Roxygen function documentation - is this something that needs to be added?
3. The Dockerfile still takes `mrc-ide/drat` as a repo, I can change this once we set up an R-universe;
4. The R CMD check GH workflow allows `NOTE`s to pass - this because {daedalus} is imported but not used (this will likely change?)

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# NOTE: manual install stringi to link against libicu70 over libicu66
+RUN install2.r --error stringi
+
+# Without this, we are unable to pick up more recent packages
+COPY docker/Rprofile.site /usr/local/lib/R/etc/Rprofile.site
+
+# More cache-friendly installation of some dependencies:
+# NOTE: need to convert to using r-universe instead
+RUN install2.r --error \
+        --repos=https://mrc-ide.github.io/drat \
```
Comment (richfitz) [inline docker/Dockerfile:21]: Can you update this

```suggestion
        --repos=https://mrc-ide.r-universe.dev \
```

everything should just work with that in place

**Review by richfitz — CHANGES_REQUESTED** [review]
tiny things, then let's get this merged

**Comment by richfitz** [comment]
Your queries:

> I'm not having much luck updating to rocker/r-ver:4.4.1 as {R6} and {jsonlite} aren't installing which blocks {porcelain};

this is now fixed :tada: 

> There's little to no Roxygen function documentation - is this something that needs to be added?

that's fine, roxygen is just a tool for us to generate the interface for now

> The Dockerfile still takes mrc-ide/drat as a repo, I can change this once we set up an R-universe;

this can be changed now, and you can also set up a jameel-institute one to pull daedalus itself in (now or later as you prefer)

> The R CMD check GH workflow allows NOTEs to pass - this because {daedalus} is imported but not used (this will likely change?)

I never block on notes, because of notes like the "more than 5mb of compiled C++" one that blocks just about everything using Rcpp/cpp11

File: R/util.R [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
-  if (is.null(x)) y else x
+scalar <- function(x) {
+  jsonlite::unbox(x)
+}
+
+package_version_string <- function(name) {
+  as.character(utils::packageVersion(name))
+}
+
+uglify <- function(code) {
```
[the remark this answers is not in the document]
Comment (pratikunterwegs) [reply R/util.R]: Removed in https://github.com/jameel-institute/daedalus.api/pull/1/commits/792b3e75afc0d530be478a93cf590e73f9f57d5e - wasn't sure whether these would be needed later so kept them initially

File: R/porcelain.R [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Generated by porcelain: do not edit by hand
```
[the remark this answers is not in the document]
Comment (pratikunterwegs) [reply R/porcelain.R:1]: Added in https://github.com/jameel-institute/daedalus.api/pull/1/commits/792b3e75afc0d530be478a93cf590e73f9f57d5e

File: R/util.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (richfitz) [reply R/util.R]: These were for uglifying js code returned from odin, but we removed the implementation in the end as it was hard to get it working as expected so I think I left it in as an empty skeleton and then forgot about it...

**Comment by pratikunterwegs** [comment]
> this can be changed now, and you can also set up a jameel-institute one to pull daedalus itself in (now or later as you prefer)

I guess this can be left for now as {daedalus} is pulled direct from GH anyway as the remote is listed.

**Comment by richfitz** [comment]
> I guess this can be left for now as {daedalus} is pulled direct from GH anyway as the remote is listed.

this is fine until we hit rate limits, as we can't get the token safely into the builder. if/once we hit that the fix is easy and by then we'll probably have the universe set up

**Review by richfitz — APPROVED** [review]
fabulous, thanks

**Comment by pratikunterwegs** [comment]
> this is fine until we hit rate limits, as we can't get the token safely into the builder. if/once we hit that the fix is easy and by then we'll probably have the universe set up

I set it up yesterday/the day before - works alright. I can remove it from remotes and add the R-universe instead to the Dockerfile.

**Pull request merged**